src/core/stores/VNode.ts

```diff
@@ -453,6 +453,32 @@ export class VNode {
             return next;
         }
     }
+    /**
+     * Return the previous leaf in a depth-first pre-order traversal of the
+     * tree that satisfies the given predicate. If no predicate is given return
+     * the previous leaf in a depth-first pre-order traversal of the tree.
+     * A leaf is a childless node.
+     *
+     * @param [predicate]
+     */
+    previousLeaf(predicate?: Predicate): VNode {
+        return this.previous((node: VNode): boolean => {
+            return (!predicate || predicate(node)) && !node.children.length;
+        });
+    }
+    /**
+     * Return the next leaf in a depth-first pre-order traversal of the tree
+     * that satisfies the given predicate. If no predicate is given return the
+     * next leaf in a depth-first pre-order traversal of the tree.
+     * A leaf is a childless node.
+     *
+     * @param [predicate]
+     */
+    nextLeaf(predicate?: Predicate): VNode {
+        return this.next((node: VNode): boolean => {
+            return (!predicate || predicate(node)) && !node.children.length;
```

Comment:
I would test the children first, as I expect most nodes are not leaves, so checking "being a leaf" first should be more efficient in a world where people might have crafted very complicated predicates.